File: ECommarce.View/src/app/core/models/order.ts
export enum OrderStatus {
  Confirmed = "Confirmed",
  Processing = "Processing",
  Packed = "Packed",
  Shipped = "Shipped",
  Delivered = "Delivered",
}

export interface Order {
  id: number;
  orderNumber: string;
  customerName: string;
  customerPhone: string;
  shippingAddress: string;
  deliveryDetails?: string;
  subTotal: number;
  tax: number;
  shippingCost: number;
  total: number;
  itemsCount: number;
  status: OrderStatus;
  createdAt: string;
  items: OrderItem[];
}

export interface OrderItem {
  id?: number;
  productId: number;
  productName: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  color?: string;
  size?: string;
  imageUrl?: string;
}

File: ECommerce.Core/Entities/Order.cs
using System;
using System.Collections.Generic;

namespace ECommerce.Core.Entities;

public enum OrderStatus
{
    Pending,
    Confirmed,
    Processing,
    Packed,
    Shipped,
    Delivered,
    Cancelled
}

public class Order : BaseEntity
{
    public string OrderNumber { get; set; } = string.Empty; // e.g. ORD-2024-001
    
    // Customer Info (Snapshot)
    public string CustomerName { get; set; } = string.Empty;
    public string CustomerPhone { get; set; } = string.Empty;
    public string ShippingAddress { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Area { get; set; } = string.Empty;
    
    // Financials
    public decimal SubTotal { get; set; }
    public decimal Tax { get; set; }
    public decimal ShippingCost { get; set; }
    public decimal Total { get; set; }
    
    public int? DeliveryMethodId { get; set; }
    public DeliveryMethod? DeliveryMethod { get; set; }
    
    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    
    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
    
    public long? SteadfastConsignmentId { get; set; }
    public string? SteadfastTrackingCode { get; set; }
    public string? SteadfastStatus { get; set; }
    public string? CreatedIp { get; set; }
}


public class OrderItem : BaseEntity
{
    public int OrderId { get; set; }
    public Order? Order { get; set; }
    
    public int ProductId { get; set; }
    public Product? Product { get; set; }
    
    public string ProductName { get; set; } = string.Empty;
    public string? Color { get; set; }
    public string? Size { get; set; }
    public string? ImageUrl { get; set; } // Snapshot of product image
    
    public decimal UnitPrice { get; set; }
    public int Quantity { get; set; }
    public decimal TotalPrice => UnitPrice * Quantity;
}
